testsuite/tests/typing-modules/applicative_functor_type.ml

```diff
@@ -43,15 +47,19 @@ type t = F(M).t
 Line 1, characters 9-15:
 1 | type t = F(M).t
              ^^^^^^
-Error: The type of M does not match F's parameter
-       Modules do not match:
-         sig type t = M.t val equal : 'a -> 'a -> bool end
-       is not included in
-         sig type t = M.t val equal : unit end
-       Values do not match:
-         val equal : 'a -> 'a -> bool
-       is not included in
-         val equal : unit
+Error: The functor application F(M) is ill-typed.
+       These arguments:
+         M
+       do not match these parameters:
+         functor (X : ...(X)) -> ...
+     Modules do not match:
+       M : sig type t = M.t val equal : 'a -> 'a -> bool end
+     is not included in
+       ...(X) = sig type t = M.t val equal : unit end
```

Comment:
I do not like this `...(X)` notation very much.
Couldn't we just reuse the name of the argument when available (or synthesize one)?
```ocaml
X : sig type t = M.t val equal : unit end
```

Reply:
That's exactly what happens: we reuse the name if we can, otherwise, if we need to name a complex signature, we synthesize a name, often by using the parameter.

We went through various ways of naming these signatures, but none of them felt exactly right, as we need to be very clear about the fact that this name doesn't exist in the code (otherwise, the error message is very confusing). `...(NAME)` is the best we came up with.

Reply:
We tried `$number`, which was mostly unreadable. We can switch back to `$name` while keeping the same way of choosing names. I'm not very convinced it's much better, but at this point I think we need to try in the wild and gather feedback.

Reply:
We have now switched to `$name` and remove cases where it introduces inconsistencies. I think further improvement/refinements should wait until we have more feedback.